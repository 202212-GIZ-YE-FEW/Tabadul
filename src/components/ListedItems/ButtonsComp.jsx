import React from "react";

import {
    AddItem,
    ButtonDetailDiv,
    ButtonIcon,
    ButtonsDiv,
    FilterButton,
    GreenButtonsDiv,
    InnerDiv,
    MenuItem,
    MenuItemDiv,
    MenuList,
} from "./ListedItems.styled";

const ButtonsComp = (props) => {
    const { setDropDown, dropDown, locationList, sortRecent, setHandleFilter } =
        props;

    return (
        <ButtonsDiv>
            <GreenButtonsDiv>
                <FilterButton onClick={() => setDropDown(!dropDown)}>
                    <InnerDiv>
                        <ButtonDetailDiv>Location</ButtonDetailDiv>
                        {dropDown && (
                            <MenuList>
                                {locationList.map((location, index) => {
                                    return (
                                        <MenuItem
                                            key={index}
                                            onClick={() => {
                                                setHandleFilter(location);
                                            }}
                                        >
                                            <MenuItemDiv>
                                                {location}
                                            </MenuItemDiv>
                                        </MenuItem>
                                    );
                                })}
                            </MenuList>
                        )}
                        <ButtonDetailDiv>
                            <ButtonIcon
                                src='/SearchFilterIcons/Polygon.svg'
                                alt='icon'
                                width={21}
                                height={19}
                            />
                        </ButtonDetailDiv>
                    </InnerDiv>
                </FilterButton>
                <FilterButton onClick={sortRecent}>
                    <InnerDiv>
                        <ButtonDetailDiv>Recent</ButtonDetailDiv>

                        <ButtonDetailDiv>
                            <ButtonIcon
                                src='/SearchFilterIcons/Recent.svg'
                                alt='icon'
                                width={35}
                                height={25}
                            />
                        </ButtonDetailDiv>
                    </InnerDiv>
                </FilterButton>
            </GreenButtonsDiv>
            <div>
                <AddItem>Add item</AddItem>
            </div>
        </ButtonsDiv>
    );
};

export default ButtonsComp;
