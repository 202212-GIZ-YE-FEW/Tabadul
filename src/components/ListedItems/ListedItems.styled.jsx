import styled from "@emotion/styled";
import Image from "next/image";
import Link from "next/link";

export const ListedItemsDiv = styled.div`
    width: 100%;
    background-color: #f1f1f1;
    display: flex;
    flex-direction: column;
    align-items: center;
`;
export const SearchAndFilterDiv = styled.div`
    width: 100%;
    display: flex;
    flex-wrap: wrap;
    gap: 5px 5%;
    justify-content: center;
`;

export const SearchDiv = styled.div`
    width: 516px;
    display: flex;
    background-color: red;
    position: relative;
    margin: 20px 0 20px 0;
    padding-inline-start: 39px;
    padding-inline-end: 39px;
    background-color: red;
    color: #1a4b38;
    display: flex;
    justify-content: space-between;
    height: 54px;
    border: none;
    background: #ffffff;
    box-shadow: 0px 5px 6px rgba(0, 0, 0, 0.25);
    border-radius: 69px;
    transition: ease all 0.5s;

    @media screen and (max-width: 1024px) {
        margin: 20px 2vw 20px 2vw;
    }
`;

export const SearchInput = styled.input`
    width: 100%;
    border: none;
    font-size: 24px;
    &:focus-visible {
        outline: none;
        border: 2px solid #33956d;
    }
    @media screen and (max-width: 600px) {
        &::placeholder {
            font-size: 18px;
        }
    }
`;

export const SearchIcon = styled(Image)`
    color: #5a5a5a;
    height: 100%;
    @media screen and (max-width: 600px) {
        width: 20px;
    }
`;

export const ButtonsDiv = styled.div`
    display: flex;
    flex-direction: row;
    justify-content: center;
    flex-wrap: wrap;
    padding-top: 6px;
    gap: 52px;
    margin: 20px;
`;

export const FilterButton = styled.button`
    position: relative;
    display: flex;
    justify-content: space-between;
    font-size: 21px;
    font-weight: 700;
    height: 56px;
    background-color: #33956d;
    color: white;
    border: none;
    border-radius: 20px;
    cursor: pointer;
    transition: ease all 0.5s;
    gap: 20px;

    @media screen and (max-width: 600px) {
        font-size: 15px;
    }
    @media screen and (max-width: 1000px) {
        font-size: 17px;
    }
`;

export const MenuList = styled.ul`
    background-color: white;
    position: absolute;
    list-style-type: none;
    top: 46px;
    right: 0px;
    padding-left: 0;
    padding-right: 0;
    box-shadow: 0px 0px 7px rgba(0, 0, 0, 0.25);
    border-radius: 5px;
`;
export const MenuItemDiv = styled.div`
    text-align: start;
    margin-right: 20px;
    margin-left: 20px;
    border-bottom: 1px solid #dfe0e0;
    padding: 15px 3px 15px 3px;
`;

export const MenuItem = styled.li`
    text-align: left;
    color: #33956d;
    font-weight: 400;

    &:last-child div {
        border-style: none;
    }
    &:hover {
        background-color: #33956d;
        color: white;
    }

    &:first-of-type:hover {
        border-top-left-radius: inherit;
        border-top-right-radius: inherit;
    }

    &:last-of-type:hover {
        border-bottom-left-radius: inherit;
        border-bottom-right-radius: inherit;
    }
`;

export const InnerDiv = styled.div`
    display: flex;
    position: relative;
    justify-content: space-between;
    padding: 0 17px 0 17px;
    height: 100%;
    width: 100%;
    gap: 10px;
`;

export const GreenButtonsDiv = styled.div`
    display: flex;
    justify-content: space-between;
    gap: 10px;
`;

export const ButtonDetailDiv = styled.div`
    display: flex;
    justify-content: center;
    flex-direction: column;
`;
export const ButtonName = styled.div`
    display: flex;
    place-items: center;
`;
export const ButtonIcon = styled(Image)`
    margin: 5px 10px 0 10px;

    @media screen and (max-width: 600px) {
        width: 17px;
    }
`;
export const AddItem = styled.button`
    font-size: 24px;
    font-weight: 700;
    height: 56px;
    color: #f07167;
    background-color: transparent;
    padding: 12px 24px;
    border: 1px solid #f07167;
    border-radius: 20px;
    cursor: pointer;
    transition: ease all 0.5s;

    @media screen and (max-width: 600px) {
        font-size: 15px;
    }
    @media screen and (max-width: 1000px) {
        font-size: 17px;
    }
`;

export const ItemsListDiv = styled.div`
    margin: 120px 0 80px 0;
    max-width: 1650px;
    width: 100%;
    height: 100%;
    display: grid;
    grid-gap: 43px 50px;
    justify-content: center;
    grid-template-columns: repeat(auto-fit, 275px);
`;

export const CardImage = styled(Image)`
    object-fit: cover;
    object-position: center top;
    border-radius: 19px 19px 0px 0px;
    width: 100%;
    height: 244px;
`;

export const ItemCard = styled.div`
    /* background-color: red; */
    width: 100%;
    height: 100%;
    background: #ffffff;
    box-shadow: 0px 0px 7px rgba(0, 0, 0, 0.25);
    border-radius: 20px;
    /* background: #ffffff; */
    margin: 10px;
    padding: 4px 4px 0 4px;
`;

export const ProductDetail = styled.div`
    padding: 0 13px 0 13px;
    /* height: 100%; */
`;

export const ItemIcon = styled(Image)`
    /* margin-left: 5px; */
`;

export const Cardinfo = styled.div`
    display: flex;
    align-items: center;
`;

export const IconsTitles = styled.p`
    font-size: 16px;
    line-height: 30px;
    color: #3c4347;
    margin: 0 5px 0 7px;
`;
export const DateAndTime = styled.p`
    /* padding: 0 0 0 12px; */
    margin: 0 0 10px 12px;
    color: #8b8b8b;
    font-size: 13px;
`;
export const ItemTitle = styled.p`
    font-weight: 600;
    font-size: 19px;
    color: #3c4347;
    margin: 0 0 0px 10px;
    word-wrap: break-word;
`;
export const LocationCategoryDiv = styled.div`
    display: flex;
    flex-direction: row;
    justify-content: start;
    gap: 10px;
    padding: 9px 0px 2px 10px;
`;
export const CardLink = styled(Link)`
    text-decoration: none;
`;

export const ShimmerContainer = styled.div`
    width: 80%;
    margin: 100px auto;
`;
