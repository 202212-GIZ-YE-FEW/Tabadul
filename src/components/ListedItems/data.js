export const items = [
    {
        id: 0,
        location: "Sana'a",
        image: "https://www.greenqueen.com.hk/wp-content/uploads/2021/07/Rental-Fashion-Causes-More-Emissions-Than-Throwing-Clothes-Away.jpg",
        category: "Clothes",
        description:
            "lgsdgsdgsdgls dglsdjglsdjglsd sdlgsdlgldslsdglksdg sdlgsdgsdlglsdlds ",
        time: "8:00:15",
        date: "2/30/2000",
    },
    {
        id: 1,
        location: "Aden",
        image: "https://www.greenqueen.com.hk/wp-content/uploads/2021/07/Rental-Fashion-Causes-More-Emissions-Than-Throwing-Clothes-Away.jpg",
        category: "Games",
        description:
            "lgsdgsdgsdgls dglsdjglsdjglsd sdlgsdlgldslsdglksdg sdlgsdgsdlglsdlds ",
        time: "6:00:15",
        date: "3/05/2023",
    },
    {
        id: 2,
        location: "Taiz",
        image: "https://www.greenqueen.com.hk/wp-content/uploads/2021/07/Rental-Fashion-Causes-More-Emissions-Than-Throwing-Clothes-Away.jpg",
        category: "Books",
        description:
            "lgsdgsdgsdgls dglsdjglsdjglsd sdlgsdlgldslsdglksdg sdlgsdgsdlglsdlds ",
        time: "3:00:15",
        date: "3/30/2023",
    },
    {
        id: 3,
        location: "Sana'a",
        image: "https://www.greenqueen.com.hk/wp-content/uploads/2021/07/Rental-Fashion-Causes-More-Emissions-Than-Throwing-Clothes-Away.jpg",
        category: "Shoes",
        description:
            "lgsdgsdgsdglsdg lsdjglsdjglsd sdlgsdlgldslsdglksdg sdlgsdgsdlglsdlds ",
        time: "11:00:15",
        date: "3/30/2022",
    },
    {
        id: 4,
        location: "Sana'a",
        image: "https://www.greenqueen.com.hk/wp-content/uploads/2021/07/Rental-Fashion-Causes-More-Emissions-Than-Throwing-Clothes-Away.jpg",
        category: "Shoes",
        description:
            "lgsdgsdgsdglsdg lsdjglsdjglsd sdlgsdlgldslsdglksdg sdlgsdgsdlglsdlds ",
        time: "12:00:15",
        date: "3/30/2023",
    },
    {
        id: 5,
        location: "Sana'a",
        image: "https://www.greenqueen.com.hk/wp-content/uploads/2021/07/Rental-Fashion-Causes-More-Emissions-Than-Throwing-Clothes-Away.jpg",
        category: "Shoes",
        description:
            "lgsdgsdgsdglsdg lsdjglsdjglsd sdlgsdlgldslsdglksdg sdlgsdgsdlglsdlds ",
        time: "11:00:15",
        date: "3/31/2023",
    },
    {
        id: 6,
        location: "Hadramout",
        image: "https://www.greenqueen.com.hk/wp-content/uploads/2021/07/Rental-Fashion-Causes-More-Emissions-Than-Throwing-Clothes-Away.jpg",
        category: "Shoes",
        description:
            "lgsdgsdgsdglsdg lsdjglsdjglsd sdlgsdlgldslsdglksdg sdlgsdgsdlglsdlds ",
        time: "11:00:15",
        date: "3/30/2023",
    },
];
