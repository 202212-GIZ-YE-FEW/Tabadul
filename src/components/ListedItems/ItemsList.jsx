import React from "react";

import { ItemsListDiv } from "./ListedItems.styled";
import Card from "../Card/Card";

const ItemsList = ({ filters }) => {
    return (
        <ItemsListDiv>
            {filters?.map((item) => {
                return (
                    <Card
                        key={item.id}
                        productImage={item.image[0].url}
                        id={item.id}
                        title={item.title}
                        location={item.location}
                        category={item.category}
                        date={item.date}
                        description={item.description}
                    />
                );
            })}
        </ItemsListDiv>
    );
};

export default ItemsList;
