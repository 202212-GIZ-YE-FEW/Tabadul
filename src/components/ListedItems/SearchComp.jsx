import React from "react";

import { SearchDiv, SearchIcon, SearchInput } from "./ListedItems.styled";

const SearchComp = ({ handleInputChange }) => {
    return (
        <SearchDiv>
            <SearchInput
                type='text'
                onChange={(e) => handleInputChange(e)}
                placeholder='Search'
            />
            <SearchIcon
                src='/SearchFilterIcons/Search.svg'
                alt='Search Icon'
                width={25}
                height={35}
            />
        </SearchDiv>
    );
};

export default SearchComp;
