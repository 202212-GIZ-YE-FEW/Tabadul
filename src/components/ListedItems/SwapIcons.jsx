import React from "react";

import { SwapIcon } from "./ListedItems.styled";

const SwapIcons = ({ type }) => {
    return (
        <>
            {type === "Recent" ? (
                <SwapIcon>
                    <svg
                        width='25'
                        height='35'
                        viewBox='0 0 46 26'
                        xmlns='http://www.w3.org/2000/svg'
                    >
                        <path
                            fill='currentcolor'
                            d='M45.6463 23.2905C45.6463 22.834 45.449 22.3962 45.0978 22.0733C44.7467 21.7505 44.2703 21.5692 43.7737 21.5692H32.5376C32.041 21.5692 31.5646 21.7505 31.2134 22.0733C30.8622 22.3962 30.6649 22.834 30.6649 23.2905C30.6649 23.747 30.8622 24.1849 31.2134 24.5077C31.5646 24.8305 32.041 25.0119 32.5376 25.0119H43.7737C44.2703 25.0119 44.7467 24.8305 45.0978 24.5077C45.449 24.1849 45.6463 23.747 45.6463 23.2905ZM45.6463 12.9625C45.6463 12.5059 45.449 12.0681 45.0978 11.7453C44.7467 11.4225 44.2703 11.2411 43.7737 11.2411H17.5562C17.0596 11.2411 16.5832 11.4225 16.232 11.7453C15.8808 12.0681 15.6835 12.5059 15.6835 12.9625C15.6835 13.419 15.8808 13.8568 16.232 14.1796C16.5832 14.5025 17.0596 14.6838 17.5562 14.6838H43.7737C44.2703 14.6838 44.7467 14.5025 45.0978 14.1796C45.449 13.8568 45.6463 13.419 45.6463 12.9625ZM45.6463 2.63443C45.6463 2.1779 45.449 1.74007 45.0978 1.41726C44.7467 1.09444 44.2703 0.913086 43.7737 0.913086H2.57482C2.07816 0.913086 1.60184 1.09444 1.25064 1.41726C0.899448 1.74007 0.702148 2.1779 0.702148 2.63443C0.702148 3.09095 0.899448 3.52878 1.25064 3.8516C1.60184 4.17441 2.07816 4.35577 2.57482 4.35577H43.7737C44.2703 4.35577 44.7467 4.17441 45.0978 3.8516C45.449 3.52878 45.6463 3.09095 45.6463 2.63443Z'
                        />
                    </svg>
                </SwapIcon>
            ) : (
                <SwapIcon>
                    <svg
                        width='20'
                        height='30'
                        viewBox='0 0 14 12'
                        xmlns='http://www.w3.org/2000/svg'
                    >
                        <path
                            fill='currentcolor'
                            d='M0 0.75C0 0.551088 0.0790177 0.360322 0.21967 0.21967C0.360322 0.0790175 0.551088 0 0.75 0H13.25C13.4489 4.19176e-09 13.6397 0.0790175 13.7803 0.21967C13.921 0.360322 14 0.551088 14 0.75C14 0.948912 13.921 1.13968 13.7803 1.28033C13.6397 1.42098 13.4489 1.5 13.25 1.5H0.75C0.551088 1.5 0.360322 1.42098 0.21967 1.28033C0.0790177 1.13968 0 0.948912 0 0.75ZM0 5.75C0 5.55109 0.0790177 5.36032 0.21967 5.21967C0.360322 5.07902 0.551088 5 0.75 5H13.25C13.4489 5 13.6397 5.07902 13.7803 5.21967C13.921 5.36032 14 5.55109 14 5.75C14 5.94891 13.921 6.13968 13.7803 6.28033C13.6397 6.42098 13.4489 6.5 13.25 6.5H0.75C0.551088 6.5 0.360322 6.42098 0.21967 6.28033C0.0790177 6.13968 0 5.94891 0 5.75ZM0.75 10H13.25C13.4489 10 13.6397 10.079 13.7803 10.2197C13.921 10.3603 14 10.5511 14 10.75C14 10.9489 13.921 11.1397 13.7803 11.2803C13.6397 11.421 13.4489 11.5 13.25 11.5H0.75C0.551088 11.5 0.360322 11.421 0.21967 11.2803C0.0790177 11.1397 0 10.9489 0 10.75C0 10.5511 0.0790177 10.3603 0.21967 10.2197C0.360322 10.079 0.551088 10 0.75 10Z'
                        />
                    </svg>
                </SwapIcon>
            )}
        </>
    );
};

export default SwapIcons;
