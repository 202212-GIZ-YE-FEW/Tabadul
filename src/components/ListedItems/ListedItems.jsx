import React, { useEffect, useState } from "react";
import { ShimmerPostList } from "react-shimmer-effects-18";

import { fetchLocations } from "@/utils/firebase";

import ButtonsComp from "./ButtonsComp";
import ItemsList from "./ItemsList";
import {
    ListedItemsDiv,
    SearchAndFilterDiv,
    ShimmerContainer,
} from "./ListedItems.styled";
import SearchComp from "./SearchComp";

const ListedItems = (props) => {
    const { setFilters, filters, setHandleFilter, handleFilter, items } = props;
    const [locationList, setLocationList] = useState();
    const [dropDown, setDropDown] = useState(false);

    function handleInputChange(e) {
        setHandleFilter(e.target.value);
    }

    function sortRecent() {
        const sortedItems = [...items];
        sortedItems.sort((a, b) => {
            const aTime = a.time.split(" ");
            const bTime = b.time.split(" ");
            return (
                b.date.localeCompare(a.date) || bTime[0].localeCompare(aTime[0])
            );
        });

        setFilters(sortedItems);
    }

    async function getLocations() {
        const locs = await fetchLocations();
        setLocationList(locs);
    }

    useEffect(() => {
        getLocations();
        if (handleFilter) {
            const filteredItems = items?.filter((item) => {
                return (
                    item.location.name
                        .toLowerCase()
                        .includes(handleFilter?.toLowerCase()) ||
                    item.category.name
                        .toLowerCase()
                        .includes(handleFilter?.toLowerCase()) ||
                    item.title
                        .toLowerCase()
                        .includes(handleFilter?.toLowerCase())
                );
            });
            setFilters(filteredItems);
        } else {
            setFilters(items);
        }
    }, [handleFilter]);

    return (
        <ListedItemsDiv>
            <SearchAndFilterDiv>
                <SearchComp
                    handleFilter={handleFilter}
                    handleInputChange={handleInputChange}
                />
                <ButtonsComp
                    setDropDown={setDropDown}
                    dropDown={dropDown}
                    locationList={locationList}
                    sortRecent={sortRecent}
                    setFilters={setFilters}
                    items={items}
                />
            </SearchAndFilterDiv>
            {filters?.length > 0 ? (
                <ItemsList filters={filters} />
            ) : (
                <ShimmerContainer>
                    <ShimmerPostList
                        postStyle='STYLE_FOUR'
                        col={4}
                        row={2}
                        gap={30}
                        rounded={true}
                    />
                </ShimmerContainer>
            )}
        </ListedItemsDiv>
    );
};

export default ListedItems;
