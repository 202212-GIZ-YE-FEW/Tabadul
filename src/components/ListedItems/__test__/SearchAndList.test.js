import renderer from "react-test-renderer";

import ListedItems from "../ListedItems";

it("renders correctly", () => {
    const tree = renderer.create(<ListedItems />).toJSON();
    expect(tree).toMatchSnapshot();
});
