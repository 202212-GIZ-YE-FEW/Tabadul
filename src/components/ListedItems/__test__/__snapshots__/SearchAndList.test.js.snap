// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="css-rsoya5"
>
  <div
    className="css-1ucdenm"
  >
    <div
      className="css-1rpnq9q"
    >
      <input
        className="css-ht1ag6"
        onChange={[Function]}
        placeholder="Search"
        type="text"
      />
      <img
        alt="Search Icon"
        className="css-aeci7b"
        data-nimg="1"
        decoding="async"
        height={35}
        loading="lazy"
        onError={[Function]}
        onLoad={[Function]}
        src="/SearchFilterIcons/Search.svg"
        style={
          Object {
            "color": "transparent",
          }
        }
        width={25}
      />
    </div>
    <div
      className="css-n40r9z"
    >
      <div
        className="css-1eb9gkv"
      >
        <button
          className="css-1gla2lw"
          onBlur={[Function]}
          onClick={[Function]}
        >
          <div
            className="css-zx4cze"
          >
            <div
              className="css-15i7kcb"
            >
              Location
            </div>
            <div
              className="css-15i7kcb"
            >
              <img
                alt="icon"
                className="css-1xxk55y"
                data-nimg="1"
                decoding="async"
                height={19}
                loading="lazy"
                onError={[Function]}
                onLoad={[Function]}
                src="/SearchFilterIcons/Polygon.svg"
                style={
                  Object {
                    "color": "transparent",
                  }
                }
                width={21}
              />
            </div>
          </div>
        </button>
        <button
          className="css-1gla2lw"
          onClick={[Function]}
        >
          <div
            className="css-zx4cze"
          >
            <div
              className="css-15i7kcb"
            >
              Recent
            </div>
            <div
              className="css-15i7kcb"
            >
              <img
                alt="icon"
                className="css-1xxk55y"
                data-nimg="1"
                decoding="async"
                height={25}
                loading="lazy"
                onError={[Function]}
                onLoad={[Function]}
                src="/SearchFilterIcons/Recent.svg"
                style={
                  Object {
                    "color": "transparent",
                  }
                }
                width={35}
              />
            </div>
          </div>
        </button>
      </div>
      <div>
        <a
          href="/addToItem"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onTouchStart={[Function]}
        >
          <button
            className="css-w0vzl6"
          >
            Add item
          </button>
        </a>
      </div>
    </div>
  </div>
  <div
    className="css-d0uswh"
  >
    <div
      className="grid grid-gap-30 grid-column-3"
    >
      <div>
        <div
          className="shimmer-card"
        >
          <div
            className=""
          >
            <div
              className="shimmer shimmer-thumbnail mb-0"
              style={
                Object {
                  "height": "250px",
                }
              }
            />
          </div>
          <div
            className="p-30"
          >
            <div
              className="grid shimmer-text grid-gap-10"
            >
              <div
                className="shimmer shimmer-text-line"
              />
              <div
                className="shimmer shimmer-text-line"
              />
              <div
                className="shimmer shimmer-text-line"
              />
            </div>
            <div
              className="shimmer shimmer-button shimmer-button--sm"
            />
          </div>
        </div>
      </div>
      <div>
        <div
          className="shimmer-card"
        >
          <div
            className=""
          >
            <div
              className="shimmer shimmer-thumbnail mb-0"
              style={
                Object {
                  "height": "250px",
                }
              }
            />
          </div>
          <div
            className="p-30"
          >
            <div
              className="grid shimmer-text grid-gap-10"
            >
              <div
                className="shimmer shimmer-text-line"
              />
              <div
                className="shimmer shimmer-text-line"
              />
              <div
                className="shimmer shimmer-text-line"
              />
            </div>
            <div
              className="shimmer shimmer-button shimmer-button--sm"
            />
          </div>
        </div>
      </div>
      <div>
        <div
          className="shimmer-card"
        >
          <div
            className=""
          >
            <div
              className="shimmer shimmer-thumbnail mb-0"
              style={
                Object {
                  "height": "250px",
                }
              }
            />
          </div>
          <div
            className="p-30"
          >
            <div
              className="grid shimmer-text grid-gap-10"
            >
              <div
                className="shimmer shimmer-text-line"
              />
              <div
                className="shimmer shimmer-text-line"
              />
              <div
                className="shimmer shimmer-text-line"
              />
            </div>
            <div
              className="shimmer shimmer-button shimmer-button--sm"
            />
          </div>
        </div>
      </div>
      <div>
        <div
          className="shimmer-card"
        >
          <div
            className=""
          >
            <div
              className="shimmer shimmer-thumbnail mb-0"
              style={
                Object {
                  "height": "250px",
                }
              }
            />
          </div>
          <div
            className="p-30"
          >
            <div
              className="grid shimmer-text grid-gap-10"
            >
              <div
                className="shimmer shimmer-text-line"
              />
              <div
                className="shimmer shimmer-text-line"
              />
              <div
                className="shimmer shimmer-text-line"
              />
            </div>
            <div
              className="shimmer shimmer-button shimmer-button--sm"
            />
          </div>
        </div>
      </div>
      <div>
        <div
          className="shimmer-card"
        >
          <div
            className=""
          >
            <div
              className="shimmer shimmer-thumbnail mb-0"
              style={
                Object {
                  "height": "250px",
                }
              }
            />
          </div>
          <div
            className="p-30"
          >
            <div
              className="grid shimmer-text grid-gap-10"
            >
              <div
                className="shimmer shimmer-text-line"
              />
              <div
                className="shimmer shimmer-text-line"
              />
              <div
                className="shimmer shimmer-text-line"
              />
            </div>
            <div
              className="shimmer shimmer-button shimmer-button--sm"
            />
          </div>
        </div>
      </div>
      <div>
        <div
          className="shimmer-card"
        >
          <div
            className=""
          >
            <div
              className="shimmer shimmer-thumbnail mb-0"
              style={
                Object {
                  "height": "250px",
                }
              }
            />
          </div>
          <div
            className="p-30"
          >
            <div
              className="grid shimmer-text grid-gap-10"
            >
              <div
                className="shimmer shimmer-text-line"
              />
              <div
                className="shimmer shimmer-text-line"
              />
              <div
                className="shimmer shimmer-text-line"
              />
            </div>
            <div
              className="shimmer shimmer-button shimmer-button--sm"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
